### PR TITLE
feat: payout routing changes

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [x] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/PayoutRouting/PayoutCurrentActiveRouting.res [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@react.component
+let make = (~routingType: array<JSON.t>) => {
+  <div className="mt-4 flex flex-col gap-6">
+    {routingType
+    ->Array.mapWithIndex((ele, i) => {
+      let id = ele->LogicUtils.getDictFromJsonObject->LogicUtils.getString("id", "")
```
Comment (JeevaRamu0104) [inline src/screens/PayoutRouting/PayoutCurrentActiveRouting.res:6]: remove multiple use of LogicUtils

File: src/screens/PayoutRouting/PayoutRoutingStack.res [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+          } else {
+            0
+          }
+        })
+        ->Array.map(Nullable.make)
+
+      setRecords(_ => sortedHistoryRecords)
+      setScreenState(_ => PageLoaderWrapper.Success)
+
+      setScreenState(_ => PageLoaderWrapper.Success)
```
Comment (JeevaRamu0104) [inline src/screens/PayoutRouting/PayoutRoutingStack.res:74]: why two setScreenState

File: src/screens/PayoutRouting/PayoutRoutingConfigure.res [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          routingRuleId=id
+          isActive
+          setCurrentRouting
+          connectorList
+          urlEntityName=PAYOUT_ROUTING
+          baseUrlForRedirection
+        />
+      | DEFAULTFALLBACK =>
+        <DefaultRouting urlEntityName=PAYOUT_DEFAULT_FALLBACK baseUrlForRedirection />
+      | _ => <> </>
```
Comment (JeevaRamu0104) [inline src/screens/PayoutRouting/PayoutRoutingConfigure.res:59]: do we need this?

File: src/screens/PayoutRouting/PayoutRoutingConfigure.res [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  let connectorList =
+    HyperswitchAtom.connectorListAtom
+    ->Recoil.useRecoilValueFromAtom
+    ->filterConnectorList(~retainInList=PayoutConnector)
+
+  let baseUrlForRedirection = "/payoutrouting"
+
+  React.useEffect1(() => {
+    let searchParams = url.search
+    let filtersFromUrl = LogicUtils.getDictFromUrlSearchParams(searchParams)->Dict.get("id")
```
Comment (JeevaRamu0104) [inline src/screens/PayoutRouting/PayoutRoutingConfigure.res:19]: open LogicUtils once and reuse it

File: src/screens/PayoutRouting/PayoutRoutingStack.res [hunk 5]
```diff
@@ -0,0 +1,154 @@
+open APIUtils
```
Comment (JeevaRamu0104) [inline src/screens/PayoutRouting/PayoutRoutingStack.res:1]: move this open statement under make

File: src/screens/PayoutRouting/PayoutRoutingConfigure.res [hunk 6]
```diff
@@ -0,0 +1,63 @@
+open RoutingTypes
+open RoutingUtils
```
Comment (JeevaRamu0104) [inline src/screens/PayoutRouting/PayoutRoutingConfigure.res:2]: move this open statement under make

File: src/screens/PayoutRouting/PayoutRoutingConfigure.res (hunk 3 above)
Comment (Riddhiagrawal001) [reply src/screens/PayoutRouting/PayoutRoutingConfigure.res:59]: Yes , as there's one more enum called 'NO_ROUTING'

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**